packages/nx/src/utils/package-json.ts

```diff
@@ -69,3 +73,42 @@ export function buildTargetFromScript(
     },
   };
 }
+
+export function readModulePackageJson(
+  moduleSpecifier: string,
+  requirePaths = [workspaceRoot]
+): {
+  packageJson: PackageJson;
+  path: string;
+} {
+  let packageJsonPath: string;
+  try {
+    packageJsonPath = require.resolve(`${moduleSpecifier}/package.json`, {
+      paths: requirePaths,
+    });
+  } catch {
+    const entryPoint = require.resolve(moduleSpecifier, {
+      paths: requirePaths,
+    });
+    let moduleRootPath = dirname(entryPoint);
+    packageJsonPath = join(moduleRootPath, 'package.json');
+
+    while (!existsSync(packageJsonPath)) {
+      moduleRootPath = dirname(moduleRootPath);
+      packageJsonPath = join(moduleRootPath, 'package.json');
+    }
+  }
+
+  const packageJson = readJsonFile(packageJsonPath);
+
+  if (!(packageJson.name === moduleSpecifier)) {
+    throw new Error(
```

Comment:
This **_shouldn't_** happen. This would occur if for some reason require.resolve(`packageName`) pointed to a folder that didn't have a moduleName. Its a safety guard, but thats it.